core/src/main/java/com/linecorp/armeria/server/throttling/RequestThrottlingMapping.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.throttling;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * Returns a {@link RequestThrottlingStrategy} instance from remote invocation parameters.
+ */
+@FunctionalInterface
+public interface RequestThrottlingMapping {
```

Comment:
I'm curious why we need this interface. Couldn't we just add `ServiceRequestContext` to `RequestThrottlingStrategy.shouldThrottle(..)` and compose strategies?